### PR TITLE
Limit Global Styles: Show pre-save notice

### PR DESCRIPTION
**Pull request by mmtr**
Fixes 829-gh-Automattic/dotcom-forge

#### Proposed Changes

Displays a notice before saving any Global Style change to inform users that it won't be public unless they upgrade the site plan.

Screen Shot 2022-09-29 at 13 20 28

#### Testing Instructions

- Apply these changes to your sandbox: `install-plugin.sh editing-toolkit update/limit-global-styles-notice`
- Go to https://wordpress.com/start and create a new free site
- Add the `wpcom-limit-global-styles` blog sticker to the new site from the Blog RC
- Sandbox the new site
- Go to Appearance > Editor
- Open the Global Styles sidebar
- Change some styles
- Click on the "Save" button
- Make sure the pre-save panel displays a notice
- Make sure the "upgrade your plan" link redirects you to the "Upgrades > Plans" page
- Make sure the "revert your styles" link resets the styles to default

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/notice.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Notice } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const GlobalStylesNotice = () => {
+	return (
+		<Notice status="warning" isDismissible={ false }>
+			{ __(
+				"Your style changes won't be public until you upgrade your plan. Uou can revert your styles.",
```
Comment (rcrdortiz) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/notice.js]: Typo change `Uou` to `You`.

Comment (rcrdortiz) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/notice.js:8]: This code is also missing the `upgrade your plan` and `revert your styles` links. It's fine to add them in a follow-up.

**Comment by rcrdortiz** [comment]
The Editing ToolKit TeamCity build failed, I've restarted it for you but it failed again. This causes that when I run `install-plugin.sh editing-toolkit update/limit-global-styles-notice` the commit getting pulled is the last one that was built successfully, which is https://github.com/Automattic/wp-calypso/commit/3ed1e9bd514fe5ed188dd0566913ba8e2be524c5 so I can't test your latest changes.

On the TeamCity build report, all the tests are passing but the linter is exiting.

**Comment by mmtr** [comment]
> The Editing ToolKit TeamCity build failed

Good catch, it was caused by some linting errors that should've been fixed in the last commit.

**Comment by rcrdortiz** [comment]
I've been playing around with this for a bit. It mostly works as expected, I did find that when you change the GS settings and then click on the reset link, the sidebar is still open and shows the `Are you ready to save?` message.

I find this is a bit misleading since you can't save anything since everything got reverted, maybe we could check if the save button is disabled and close the sidebar or some other action that is more intuitive than having the sidebar open and displaying a message about saving changes and not actually letting the user save anything.

Here is a video of what I mean:

https://user-images.githubusercontent.com/1989914/193261239-a4d161c4-fc22-4b1a-ac0e-df1e9d49de5b.mp4

**Comment by mmtr** [comment]
> maybe we could check if the save button is disabled and close the sidebar

Unfortunately Gutenberg doesn't provide any mechanism for closing the sidebar – everything is handled using an internal state that it is not exposed publicly:

https://github.com/WordPress/gutenberg/blob/0b30a4cb34d39c9627b6a3795a18aee21019ce25/packages/edit-site/src/components/editor/index.js#L137-L138

So I have to add a fragile CSS selector which very likely will be broken in future releases of Gutenberg since we don't have a more specific/robust CSS selector we can use:

https://github.com/WordPress/gutenberg/blob/0b30a4cb34d39c9627b6a3795a18aee21019ce25/packages/editor/src/components/entities-saved-states/index.js#L224-L231

**Review by rcrdortiz — APPROVED** [review]

**Comment by rcrdortiz** [comment]
> Unfortunately Gutenberg doesn't provide any mechanism for closing the sidebar – everything is handled using an internal state that it is not exposed publicly:

The rest looks good to me.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7561288

Thank you @mmtr for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7561288).

**Pull request merged**